lib/ansible/modules/cloud/vmware/vmware_guest_network.py

```diff
@@ -0,0 +1,489 @@
+  delegate_to: localhost
+  register: network_facts
+'''
+
+RETURN = """
+network_data:
+    description: metadata about the virtual machine's network adapter after managing them
+    returned: always
+    type: dict
+    sample: {
+        "0": {
+            "label": "Network Adapter 1",
+            "name": "VM Network",
+            "device_type": "E1000E",
+            "mac_addr": "00:50:56:89:dc:05",
+            "unit_number": 7,
+            "wake_onlan": false,
+            "allow_guest_ctl": true,
+            "connected": true,
+            "start_connected": true,
+        },
+    }
+"""
+
+import re
+
+try:
+    from pyVmomi import vim
+except ImportError:
+    pass
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils._text import to_native, to_text
+from ansible.module_utils.vmware import PyVmomi, vmware_argument_spec, wait_for_task, find_obj, get_all_objs
+
+
+class PyVmomiHelper(PyVmomi):
+    def __init__(self, module):
+        super(PyVmomiHelper, self).__init__(module)
+        self.change_detected = False
+        self.config_spec = vim.vm.ConfigSpec()
+        self.config_spec.deviceChange = []
+        self.nic_device_type = dict(
+            pcnet32=vim.vm.device.VirtualPCNet32,
+            vmxnet2=vim.vm.device.VirtualVmxnet2,
+            vmxnet3=vim.vm.device.VirtualVmxnet3,
+            e1000=vim.vm.device.VirtualE1000,
+            e1000e=vim.vm.device.VirtualE1000e,
+            sriov=vim.vm.device.VirtualSriovEthernetCard,
+        )
+
+    def get_device_type(self, device_type=None):
+        """ Get network adapter device type """
+        nic_device_type = dict(
+            pcnet32=vim.vm.device.VirtualPCNet32(),
+            vmxnet2=vim.vm.device.VirtualVmxnet2(),
+            vmxnet3=vim.vm.device.VirtualVmxnet3(),
+            e1000=vim.vm.device.VirtualE1000(),
+            e1000e=vim.vm.device.VirtualE1000e(),
+            sriov=vim.vm.device.VirtualSriovEthernetCard(),
+        )
+        if device_type and device_type in nic_device_type:
+            return nic_device_type[device_type]
+        else:
+            self.module.fail_json(msg='Invalid network device_type %s' % device_type)
+
+    @staticmethod
+    def get_parent_datacenter(obj):
+        """ Walk the parent tree to find the object's datacenter """
+        datacenter = None
+        while True:
+            if not hasattr(obj, 'parent'):
+                break
+            obj = obj.parent
+            if isinstance(obj, vim.Datacenter):
+                datacenter = obj
+                break
+        return datacenter
+
+    def network_exists_by_name(self, content, name):
+        """ Check if network with specified name exists """
+        vimtype = [vim.Network]
+        result = find_obj(content, vimtype, name)
+        if result:
+            if to_text(self.get_parent_datacenter(result).name) != to_text(self.params['datacenter']):
+                objects = get_all_objs(content, vimtype)
```

Comment:
This will remove check for datacenter from line 215 

```suggestion
                dc_obj = self.find_datacenter_by_name(datacenter_name=self.params['datacenter'])
                if not dc_obj:
                   self.module.fail_json(msg="Failed to find the datacenter %s" % self.params['datacenter'])
                objects = get_all_objs(content, vimtype, folder=dc_obj.networkFolder)
```